divi/src/PoSTransactionCreator.cpp

```diff
@@ -180,7 +179,12 @@ bool PoSTransactionCreator::PopulateCoinstakeTransaction(
     static std::set<std::pair<const CWalletTx*, unsigned int> > setStakeCoins;
     static int nLastStakeSetUpdate = 0;
     if(!SelectCoins(allowedStakingAmount,nLastStakeSetUpdate,setStakeCoins)) return false;
-    if (GetAdjustedTime() <= chainActive.Tip()->nTime) MilliSleep(10000);
+
+    nTxNewTime = GetAdjustedTime();
+    if (nTxNewTime <= chainActive.Tip()->nTime)
+        nTxNewTime = chainActive.Tip()->nTime + 1;
+    if (nTxNewTime <= chainActive.Tip()->GetMedianTimePast())
+        nTxNewTime = chainActive.Tip()->GetMedianTimePast() + 1;
```

Comment:
Ok sure!  I'm not so familiar with the PoS changes yet, but can of course adjust it.

Reply:
Updated.  On regtest we keep the previous smaller time, so that more blocks can be generated in one go (closer together in time) before we have to adjust the mock time.